File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        min_diff, res = float("inf"), []
        for i in range(0, len(arr)-1):
            diff = arr[i+1] - arr[i]
            if diff < min_diff:
                min_diff = diff
                res = [[arr[i], arr[i+1]]]
            elif diff == min_diff:
                res.append([arr[i], arr[i+1]])
        return res

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.js
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a, b) => a - b)
    let minDiff = Infinity
    let res = []
    for (let i=0; i < arr.length-1; i++) {
        const diff = arr[i+1] - arr[i]
        if (diff < minDiff) {
            minDiff = diff
            res = [[arr[i], arr[i+1]]]
        } else if (diff === minDiff) {
            res.push([arr[i], arr[i+1]])
        }
    }
    return res
};